### PR TITLE
[Service Connector] `az webapp/spring-cloud connection create/update`: Provide `--service-endpoint` parameter to support vnet scenario

### PR DESCRIPTION
**Pull request by xfz11**
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py [hunk 1]
```diff
@@ -153,12 +156,14 @@ def add_confluent_kafka_argument(context):
...
             with self.argument_context('{} connection update {}'.format(source.value, target.value)) as c:
                 add_client_type_argument(c, source, target)
                 add_connection_name_argument(c, source)
                 add_source_resource_block(c, source)
                 add_auth_block(c, source, target)
                 add_secret_store_argument(c)
+                add_service_endpoint_argument(c)
 
         # special target resource: independent implementation
         target = RESOURCE.ConfluentKafka
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py:170]: does confluent-cloud support vnet？

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py [hunk 2]
```diff
@@ -107,6 +107,9 @@ def add_new_addon_argument(context, source, target):
     def add_secret_store_argument(context):
         context.argument('key_vault_id', options_list=['--vault-id'], help='The id of key vault to store secret value')
 
+    def add_service_endpoint_argument(context):
+        context.argument('service_endpoint', options_list=['--service-endpoint'], arg_type=get_three_state_flag(), default=None, help='Connect target service by service endpoint')
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py]: If possible, give a brief intro in the help, at whats case users can use the parameter, i.e, the requirements the source and target resource should meet

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 3]
```diff
@@ -152,6 +152,7 @@ def connection_create(cmd, client,  # pylint: disable=too-many-locals
                       user_identity_auth_info=None, system_identity_auth_info=None,
                       service_principal_auth_info_secret=None,
                       key_vault_id=None,
+                      service_endpoint=None,
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:155]: why only service endpoint? what about other vnet scenarios? private link?

**Comment by houk-ms** [comment]
PR title should follow CLI format, which will be used to generate release note, something like

[Service Connector] `az webapp/spring-cloud connection create/update`: Provider `--service-endpoint` parameter to support vnet scenario

**Comment by yonzhan** [comment]
Add @kairu-ms to help with

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py (hunk 3 above)
Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:155]: suppose there will be some description to redirect user to private link solution right

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py (hunk 1 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py:170]: no

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py (hunk 2 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/_params.py]: added

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py (hunk 3 above)
Comment (xfz11) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:155]: more description added

**Review by yungezz — APPROVED** [review]

**Comment by xfz11** [comment]
@kairu-ms Could you help to merge the PR

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 4]
```diff
@@ -235,6 +242,7 @@ def connection_update(cmd, client,
...
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:245]: This PR together with #21763 causes CI failure:

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1464664&view=logs&j=36dd4138-4d53-5e46-00d9-e5cd9744be05&t=1cf3879c-0a42-5ccd-7693-7a4781d739d8&l=94

```
src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:244:0: R0914: Too many local variables (26/25) (too-many-locals)
src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:244:0: R0914: Too many local variables (26/25) (too-many-locals)
```

**Pull request merged**